frontend/src/ui/common/EventListRow.js

```diff
@@ -0,0 +1,128 @@
+import React, {useState} from "react";
+import {useDispatch} from "react-redux";
+import {Accordion, Button} from "react-bootstrap";
+import {httpConfig} from "../../utils/httpConfig";
+import {fetchBookedMarkedEventByUserId} from "../../store/bookmarkevent";
+import {fetchEventByUserId} from "../../store/registeredeventsbyuser";
+import {dateTimeToDate, dateTimeToTime} from "../dateFormat";
+
+
+export const EventListRow = (props) => {
+    const dispatch = useDispatch();
+    const [bookmarkButtonText, setBookmarkButtonText] = useState("Bookmark");
+
+    const registerThisEvent = () => {
+        httpConfig.post(`/apis/volunteer/${props.event.eventId}`)
+            .then(reply => {
+                if(reply.status === 200) {
+                    dispatch(fetchEventByUserId())
+                }
+            })
+    }
+
+    const registerBookmarkedEvent = () => {
+        registerThisEvent();
+        removeAddBookmark();
+    }
+
+    const unRegisterEvent = () =>{
+        httpConfig.delete(`/apis/volunteer/deleteSelf/${props.event.eventId}`)
+            .then(reply => {
+                if(reply.status === 200) {
+                    dispatch(fetchEventByUserId())
+                }
+            })
+    }
+
+    const removeAddBookmark = () =>{
+        httpConfig.post(`/apis/bookmarkedEvent/${props.event.eventId}`)
+            .then(reply => {
+                if(reply.status === 200) {
+                    dispatch(fetchBookedMarkedEventByUserId());
+                    setBookmarkButtonText(() => {
```

Comment:
This hook works, but only for the event list item. May need a deeper state variable or another store call to manage the button text when using the bookmarked event section button to remove an event. Removing from the bookmarked event section directly causes the event to be removed, but the event list bookmark button text stays the same.